pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCudaBase.hpp

```diff
@@ -141,6 +155,127 @@ class TNCudaBase : public TensornetBase<Precision, Derived> {
         return dev_tag_;
     }
 
+    /**
+     * @brief Append multiple gates to the compute graph.
+     * NOTE: This function does not update the quantum state but only appends
+     * gate tensor operator to the graph.
+     * @param ops Vector of gate names to be applied in order.
+     * @param ops_wires Vector of wires on which to apply index-matched gate
+     * name.
+     * @param ops_adjoint Indicates whether gate at matched index is to be
+     * inverted.
+     */
+    void
+    applyOperations(const std::vector<std::string> &ops,
+                    const std::vector<std::vector<std::size_t>> &ops_wires,
+                    const std::vector<bool> &ops_adjoint,
+                    const std::vector<std::vector<PrecisionT>> &ops_params) {
```

Comment:
default value?
```suggestion
                    const std::vector<bool> &ops_adjoint,
                    const std::vector<std::vector<PrecisionT>> &ops_params = {}) {
```

Reply:
We can't default `ops_params` as `empty` since  it will lead the `memory out of bounds` issue at line 185.